From 8f5b1aa9025a22123c010ddff4d3642d781f0d4d Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 10:26:12 -0700
Subject: [PATCH 26/96] HBASE-7314 Can't start REST/Thrift server if HBASE_JMX_OPTS not set

Reason: Bug
Author: Jimmy Xiang
Ref: CDH-9400

git-svn-id: https://svn.apache.org/repos/asf/hbase/trunk@1419955 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 41df5319156f66daf1689f82d3bdde4090671ad8)
---
 bin/hbase-config.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/hbase-config.sh b/bin/hbase-config.sh
index c5303e8..586034d 100644
--- a/bin/hbase-config.sh
+++ b/bin/hbase-config.sh
@@ -76,7 +76,7 @@ HBASE_REGIONSERVERS="${HBASE_REGIONSERVERS:-$HBASE_CONF_DIR/regionservers}"
 # List of hbase secondary masters.
 HBASE_BACKUP_MASTERS="${HBASE_BACKUP_MASTERS:-$HBASE_CONF_DIR/backup-masters}"
 # Thrift JMX opts
-if [ -z "$HBASE_THRIFT_JMX_OPTS" ]; then
+if [[ -n "$HBASE_JMX_OPTS" && -z "$HBASE_THRIFT_JMX_OPTS" ]]; then
   HBASE_THRIFT_JMX_OPTS="$HBASE_JMX_OPTS -Dcom.sun.management.jmxremote.port=10103"
 fi
 # Thrift opts
@@ -85,7 +85,7 @@ if [ -z "$HBASE_THRIFT_OPTS" ]; then
 fi
 
 # REST JMX opts
-if [ -z "$HBASE_REST_JMX_OPTS" ]; then
+if [[ -n "$HBASE_JMX_OPTS" && -z "$HBASE_REST_JMX_OPTS" ]]; then
   HBASE_REST_JMX_OPTS="$HBASE_JMX_OPTS -Dcom.sun.management.jmxremote.port=10105"
 fi
 # REST opts
-- 
1.7.0.4

